src/extra/components/xgm.py

```diff
@@ -0,0 +1,498 @@
+        if device.is_instrument and device.source.endswith(":output"):
+            device = device.source.removesuffix(":output")
+        else:
+            device = device.source
+
+    elif isinstance(device, str):
+        if device in run.all_sources:
+            # We allow complete device names
+            pass
+        elif device in run.alias:
+            # And aliases
+            device = _find_xgm(run, run.alias[device])
+        else:
+            # And unique substrings of available XGMs
+            matches = [xgm_name for xgm_name in available_xgms
+                       if device.upper() in xgm_name]
+
+            if len(matches) == 0:
+                raise RuntimeError(f"Couldn't identify an XGM from '{device}'; please pass a valid device name, alias, or unique substring")
+            elif len(matches) == 1:
+                device = matches[0]
+            elif len(matches) > 1:
+                raise RuntimeError(f"Multiple XGMs found matching '{device}', please be more specific: {matches}")
+    else:
+        raise TypeError(f"Unrecognized `device` type '{type(device)}', must be a SourceData, KeyData, or string")
+
+    return device
+
+class XGM:
+    """Interface to an XGM (X-ray Gas Monitor).
+
+    Example usage in a Jupyter notebook:
+    ```python
+            -----------------------------------------------------------
+    In [1]: |xgm = XGM(run) # This will work with a single XGM        |
+            |xgm                                                      |
+            -----------------------------------------------------------
+    Out[1]: <XGM for SA1_XTD2_XGM/XGM/DOOCS at 9.3 KeV>
+
+            -----------------------------------------------------------
+    In [2]: |# With multiple XGMs, one needs to be chosen explicitly. |
+            |# Either by explicit device name:                        |
+            |xgm = XGM(run, "SPB_XTD9_XGM/XGM/DOOCS")                 |
+            |                                                         |
+            |# Or by passing a SourceData object:                     |
+            |xgm = XGM(run, run.alias["spb-xgm"])                     |
+            |                                                         |
+            |# Or a KeyData object:                                   |
+            |xgm = XGM(run, run.alias["spb-xgm"]["data.intensityTD"]) |
+            |                                                         |
+            |# Or alias name                                          |
+            |xgm = XGM(run, "spb-xgm")                                |
+            |                                                         |
+            |# Or a unique (case-insensitive) substring among XGMs    |
+            |xgm = XGM(run, "spb")                                    |
+            -----------------------------------------------------------
+    Out[2]: <XGM for SPB_XTD9_XGM/XGM/DOOCS at 9.3 KeV>
+    ```
```

Comment:
I see we've done this in the scan components as well, but I don't particularly like the ASCII art Jupyter notebook styling, either in the source or in the rendered docs. :shrug:

Reply:
I see, why is that? For context, the reason I made those over e.g. a Python interpreter output is because I felt that it would be familiar to more people, since most people will interact with this thing from a notebook.

Reply:
I haven't really got reasoning here, I just think it looks rather ugly, and even if people are most used to notebooks, I'd prefer to have examples presented without the ASCII art. It's not important, so if you prefer it enough, feel free to stick with it. I'm just registering that I'm not sold on it. :wink:

Reply:
But I love ASCII art :heart_eyes: I dunno, I don't feel that strongly about it but I do think it's a better example than an interpreter session so I'm inclined to stick with it.

(I spent a bit of time trying to copy the HTML from a Jupyter notebook cell into the markdown docstring, but that didn't go very well...)